piptools/repositories/local.py

```diff
@@ -56,8 +56,11 @@ def DEFAULT_INDEX_URL(self):
     def clear_caches(self):
         self.repository.clear_caches()
 
+    @contextmanager
     def freshen_build_caches(self):
-        self.repository.freshen_build_caches()
+        with super(LocalRequirementsRepository, self).freshen_build_caches():
+            with self.repository.freshen_build_caches():
+                yield
```

Comment:
```suggestion
        with self.repository.freshen_build_caches():
            yield
```

Reply:
Calling the parent implementation allows us to easily make changes to the base class and be confident all children will using it without additional intervention. If we were to add behavior to the base class's method, it is all too easy to forget one of the child implementation that must change.

Reply:
Yeah, I understand your point which I think is pretty reasonable. Although following the YAGNI principle I'd rather have an abstract method instead of a base one (which does nothing useful even for a while). Once we need to add a base logic (which I slightly doubt) we could update children's methods and use the base functionality. However, If you feel strongly that we need the base method then feel free to ignore the suggestion - I'd like to not block the PR any longer 👍

> Can you explain a bit how that helps us, what the benefit is, or what it works towards solving?

The main benefit is less code and indentations.